OpenRA.Mods.Common/Traits/Conditions/GrantConditionOnLineBuildNode.cs

```diff
@@ -0,0 +1,92 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Linq;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Grants a condition if this becomes a LineBuild corner segment.")]
+	public class GrantConditionOnLineBuildNodeInfo : ITraitInfo, Requires<LineBuildInfo>, Requires<ConditionManagerInfo>
+	{
+		[FieldLoader.Require]
+		[GrantedConditionReference]
+		[Desc("Condition to grant.")]
+		public readonly string Condition = null;
+
+		[FieldLoader.Require]
+		[Desc("Wall type for connections")]
+		public readonly string Type = "wall";
+
+		public object Create(ActorInitializer init) { return new GrantConditionOnLineBuildNode(init.Self, this); }
+	}
+
+	public class GrantConditionOnLineBuildNode : INotifyLineBuildSegmentsChanged
+	{
+		readonly Actor self;
+		readonly GrantConditionOnLineBuildNodeInfo info;
+		readonly ConditionManager conditionManager;
+
+		int conditionToken = ConditionManager.InvalidConditionToken;
+		bool wasNode;
+
+		public GrantConditionOnLineBuildNode(Actor self, GrantConditionOnLineBuildNodeInfo info)
+		{
+			this.self = self;
+			this.info = info;
+
+			conditionManager = self.Trait<ConditionManager>();
+		}
+
+		void CheckCondition()
+		{
+			var adjacentActors = CVec.Directions.SelectMany(dir =>
+				self.World.ActorMap.GetActorsAt(self.Location + dir));
+
+			var adjacent = 0;
+
+			foreach (var a in adjacentActors)
+			{
+				CVec facing;
+				var wc = a.TraitsImplementing<IWallConnector>().FirstOrDefault(Exts.IsTraitEnabled);
```

Comment:
We might want to use `FirstEnabledTraitOrDefault` (#14444) here instead. (?)

Reply:
> 01:40:40 (reaperrr) I guess you're right, though i doubt the perf gain will be of relevance (not something that is going to be called frequently)
01:41:12 (reaperrr) but since we have the helper he might as well use it, yes